renku/cli/init.py

```diff
@@ -95,97 +98,191 @@ def store_directory(ctx, param, value):
+    # preparation
     if not client.use_external_storage:
         use_external_storage = False
 
     ctx.obj = client = attr.evolve(
-        client,
-        path=directory,
-        use_external_storage=use_external_storage,
+        client, path=path, use_external_storage=use_external_storage
     )
 
-    msg = 'Initialized empty project in {path}'
-    branch_name = None
-
-    stack = contextlib.ExitStack()
-
-    if force and client.repo:
-        msg = 'Initialized project in {path} (branch {branch_name})'
-        merge_args = ['--no-ff', '-s', 'recursive', '-X', 'ours']
-
+    def print_done():
+        click.echo(' DONE ', nl=False)
+        click.echo(click.style('\u2713', fg='green'))
+
+    # verify path status
+    if len(list(client.path.glob('**/*'))) > 0:
+        if not force:
+            error = FileExistsError(
+                'Folder "{0}" is not empty. Please '
+                'add --force flag to transform it into a Renku repository.'.
+                format(str(path))
+            )
+            raise click.UsageError(error)
+        else:
+            try:
+                commit = client.find_previous_commit(
+                    str(client.renku_metadata_path),
+                )
+                branch_name = 'pre_renku_init_' + str(commit)[:8]
+                with client.worktree(
+                    branch_name=branch_name,
+                    commit=commit,
+                    merge_args=['--no-ff', '-s', 'recursive', '-X', 'ours']
+                ):
+                    click.echo(
+                        'Saving current data in branch {0}'.
+                        format(branch_name)
+                    )
+            except AttributeError:
+                click.echo('Warning! Overwriting non-empty folder.')
+            except GitCommandError as e:
+                click.UsageError(e)
+
+    # select template source
+    if template_source:
+        click.echo(
+            'Fetching template from {0}@{1}...'.format(
+                template_source, template_ref
+            ),
+            nl=False
+        )
+        template_folder = Path(mkdtemp())
         try:
-            commit = client.find_previous_commit(
-                str(client.renku_metadata_path),
+            fetch_template(template_source, template_ref, template_folder)
+            template_manifest = read_template_manifest(
+                template_folder, checkout=True
             )
-            branch_name = 'renku/init/' + str(commit)
-        except KeyError:
-            from git import NULL_TREE
-            commit = NULL_TREE
-            branch_name = 'renku/init/root'
-            merge_args.append('--allow-unrelated-histories')
-
-        ctx.obj = client = stack.enter_context(
-            client.worktree(
-                branch_name=branch_name,
-                commit=commit,
-                merge_args=merge_args,
+        except (ValueError, GitCommandError) as e:
```

Comment:
Handle this exceptions inside core module and then raise renku exceptions from them so core module raises mostly renku exceptions.